### PR TITLE
Feature/WalletConnect

### PR DESCRIPTION
**Pull request by pedrouid**

### CODE REVIEW

File: src/screens/WalletConnect/WalletConnectCallRequest.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+    const {
+      to, gasPrice, gasLimit, value, data,
+    } = payload.params[0];
+
+    let symbol = 'ETH';
+    let asset = null;
+    let amount = utils.formatEther(value).toString();
+
+    const isTokenTransfer = data.toLowerCase() !== '0x' && data.toLowerCase().startsWith('0xa9059cbb');
```
Comment (grilix) [inline src/screens/WalletConnect/WalletConnectCallRequest.js]: Can we assign a constant name to `'0xa9059cbb'` so it's clear what it means?

Comment (pedrouid) [reply src/screens/WalletConnect/WalletConnectCallRequest.js]: Fixed on [a3e5a96a6fc830db933c4d42e361d819397ccce5](https://github.com/pillarwallet/pillarwallet/pull/1095/commits/a3e5a96a6fc830db933c4d42e361d819397ccce5)

File: ios/pillarwallet.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1807,14 +1807,15 @@
 					"-ObjC",
 					"-lc++",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.pillarproject.wallet;
+				PRODUCT_BUNDLE_IDENTIFIER = "com.pillarproject.wallet-pedro";
```
Comment (grilix) [inline ios/pillarwallet.xcodeproj/project.pbxproj]: I'd say we should keep the previous value on these.

Comment (pedrouid) [reply ios/pillarwallet.xcodeproj/project.pbxproj]: Yeah, totally. That was just for me to run it on my iPhone

**Comment by grilix** [comment]
@pedrouid there are a couple of small details to fix and some conflicts to resolve before we can merge this one. Can you please take care of those so we can do a final review?

**Comment by pedrouid** [comment]
> @pedrouid there are a couple of small details to fix and some conflicts to resolve before we can merge this one. Can you please take care of those so we can do a final review?

Will do 👍

**Comment by pedrouid** [comment]
> Looking good to me.

Now that you mentioned the `Me` screen. I need to check the changes that have been merged. Give some time to check this

File: src/screens/WalletConnect/WalletConnectCallRequest.js [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    const transactionPayload = {
+      gasLimit: utils.bigNumberify(gasLimit).toNumber(),
+      amount: utils.bigNumberify(amount).toNumber(),
+      to,
+      gasPrice: utils.bigNumberify(gasPrice).toNumber(),
+      txFeeInWei: utils.bigNumberify(txFeeInWei).toNumber(),
+      symbol,
+      contractAddress: asset ? asset.address : '',
+      decimals: asset ? asset.decimals : 18,
+      node: this.state.note,
```
Comment (poocart) [inline src/screens/WalletConnect/WalletConnectCallRequest.js]: typo for `node`?

Comment (poocart) [inline src/screens/WalletConnect/WalletConnectCallRequest.js]: typo for `node`?

Comment (poocart) [reply src/screens/WalletConnect/WalletConnectCallRequest.js]: fixed

**Pull request merged**